plugins/civo/api_key.go

```diff
@@ -0,0 +1,138 @@
+package civo
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	//"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIKey,
+		DocsURL:       sdk.URL("https://www.civo.com/docs/account/api-keys"),
+		ManagementURL: sdk.URL("https://dashboard.civo.com/security"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.APIKey,
+				MarkdownDescription: "API Key used to authenticate to Civo.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 50,
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.APIKeyID,
+				MarkdownDescription: "The Name of apikey used to authenticate to civo",
+				Optional:            true,
+			},
+		},
+		DefaultProvisioner: civoProvisioner{},
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+			importer.TryEnvVarPair(secondEnvVarMapping),
+			TryCivoConfigFile("~/.civo.json"),
+		)}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"CIVO_TOKEN": fieldname.APIKey,
```

Comment:
@itsCheithanya @siddhikhapare Do I understand correctly that Civo supports "API Keys" and that the API Keys can be stored either in `CIVO_API_KEY` or `CIVO_TOKEN` environment variable?

Reply:
@arunsathiya Yes. we created it to only work with the CIVO_TOKEN option as an environmental variable in previous implementation. Now, we are working on enhancing it to support both options for obtaining API keys through environmental variables. As per @hculea's suggestion, we will need to create a custom Provisioner and importer to implement this feature.

Reply:
I would argue that we could ship this as is, as it already provides value by only provisioning through CIVO_TOKEN, and then we could work on improving the provisioning mechanism on demand, later.